app/src/main/java/nl/eduvpn/app/viewmodel/BaseConnectionViewModel.kt

```diff
@@ -90,9 +92,31 @@ abstract class BaseConnectionViewModel(
                             preferencesService.currentInstance = instance
                             preferencesService.currentDiscoveredAPI = discoveredAPI
                             preferencesService.currentAuthState = savedToken.authState
+                            preferencesService.currentProfileList = null
                             historyService.cacheAuthorizationState(instance, savedToken.authState)
                         }
-                        fetchProfiles(instance, discoveredAPI, savedToken.authState)
+                        if (preferencesService.appSettings.useWireGuard()) {
+                            val wireguardAPI = WireGuardAPI(apiService, discoveredAPI.apiBaseUri)
+                            connectionState.value = ConnectionState.CheckingServerWireGuardSupport
+                            runCatchingCoroutine {
+                                wireguardAPI.wireguardEnabled(savedToken.authState)
+                            }.onSuccess { wireguardEnabled ->
+                                if (wireguardEnabled) {
+                                    Log.d(TAG, "WireGuard enabled.")
+                                } else {
+                                    Log.d(TAG, "WireGuard not enabled.")
+                                }
+                                if (wireguard(instance, discoveredAPI, savedToken.authState)) {
```

Comment:
I think this would look cleaner this way:
```
if (wireguardEnabled) {
    Log.d(TAG, "WireGuard enabled.")
    tryConnectViaWireGuard(...) // this is the wireguard(...) method you have at line 109
} else {
   Log.d(TAG, "WireGuard not enabled.")
   fetchProfiles(...)
}
```
And you can call fetchProfiles manually from tryConnectViaWireGuard as a fallback.